### PR TITLE
feat: improve test stability by optional pricing source

### PR DESCRIPTION
**Pull request by cryptomatictrader**
This commit supports an additional environment variable `COINGECKO_API_KEY`, when specified, token price will be fetched from Coingecko service when calculating pool liquidity capitalization value during test executions.

Tested with and without the API key. Both yield the same outcome at the time of this commit.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new environment variable for fetching real-time token prices, enhancing test accuracy for liquidity calculations.
  - Added `AssetListService` for fetching and caching asset metadata, improving efficiency in accessing asset details.
  - Enhanced `CoingeckoService` with API key support for authenticated access, improving service reliability and error handling.
  - Expanded functionality in `fetch_tokens` and `fetch_pools` to include liquidity cap calculations using real-time prices.

- **Bug Fixes**
  - Improved error handling in liquidity calculations to ensure uninterrupted processing of pool data.

- **Documentation**
  - Updated README to reflect changes in the testing framework and new services introduced.

### CODE REVIEW

File: tests/data_service.py [hunk 1]
```diff
@@ -16,6 +19,19 @@ def fetch_tokens():
     except requests.exceptions.RequestException as e:
         print(f"Error fetching data from Numia: {e}")
         return []
+    
+# Helper function to calcualte token cap given the token amount 
+# and the token price from coingecko
+def get_token_cap(denom, amount_str):
```
Comment (p0mvn) [inline tests/data_service.py:25]: Let's make a note that amount must be descaled (exponent not applied)

File: tests/data_service.py [hunk 2]
```diff
@@ -33,6 +49,30 @@ def fetch_pools():
...
+                                denom = token.get('denom')
+                                token_cap = get_token_cap(denom, token['amount'])
+                                if token_cap is None:
+                                    all_token_cap_captured = False
+                                    break
+                                total_pool_cap += token_cap
+                            if all_token_cap_captured:
+                                pool.update({'liquidity': float(total_pool_cap)})
+                        except Exception as e:
+                            print(f"warning: error processing pool data (pool id {pool.get('pool_id')}) using Coingecko: {e}. fallback to numia data")
```
Comment (p0mvn) [inline tests/data_service.py]: Consider encapsulating this via a separate function

**Review by p0mvn — APPROVED** [review]
LGTM. Thanks!

File: tests/data_service.py (hunk 2 above)
Comment (cryptomatictrader) [reply tests/data_service.py]: Done 👍

**Pull request merged**